src/reanimated2/layoutReanimation/animationBuilder/Keyframe.ts

```diff
@@ -0,0 +1,199 @@
+    this.definitions = definitions;
+  }
+
+  private parseDefinitions(): ParsedKeyFramesDefinition {
+    /* 
+        for each style property contains an array with all its key points: 
+        value, duration of transition to that value, and optional easing function (defaults to Linear)
+    */
+    const parsedKeyFrames: {
+      [key: string]: KeyFramePoint[];
+    } = {};
+    const initialValues: StyleProps = this.definitions['0']
+      ? (this.definitions['0'] as StyleProps)
+      : {};
+    const duration: number = this.durationV ? this.durationV : 500;
+    const animationKeyPoints: Array<string> = Array.from(
+      Object.keys(this.definitions)
+    );
+    let previousKeyPoint = 0;
+
+    /* 
+       Because one of the assumptions is that each keyframe should contains
+       the same set of properties ( to make animation more consistent )
+       new entries in the parsedKeyFrames map can be added only in the first iteration
+       ( flag addIfNotExists is then set to true ).
+    */
+    const addKeyPoint = (
+      key: string,
+      duration: number,
+      value: string | number,
+      easing: EasingFn,
+      addIfNotExists: boolean
+    ) => {
+      if (!(key in parsedKeyFrames)) {
+        if (!addIfNotExists) {
+          throw Error(
+            'Each keyframe should contains the same set of properties!'
+          );
+        }
+        parsedKeyFrames[key] = [];
+      }
+      parsedKeyFrames[key].push({
+        duration: duration,
+        value: value,
+        easing: easing,
+      });
+    };
+    animationKeyPoints
+      .filter((value: string) => parseInt(value) !== 0)
+      .sort((a: string, b: string) => parseInt(a) - parseInt(b))
+      .forEach((keyPoint: string, index: number) => {
+        if (parseInt(keyPoint) < 0 || parseInt(keyPoint) > 100) {
+          throw Error('Keyframe should be in between range 0 - 100.');
+        }
+        const keyframe: KeyFrameProps = this.definitions[keyPoint];
+        let propsInKeyframe = 0;
+        const easing = keyframe.easing;
+        delete keyframe.easing;
+        const animationDuration =
+          ((parseInt(keyPoint) - previousKeyPoint) / 100) * duration;
+        Object.keys(keyframe).forEach((key: string) => {
+          if (key === 'transform') {
+            keyframe[key].forEach((transformStyle) => {
+              Object.keys(transformStyle).forEach((key: string) => {
+                propsInKeyframe = propsInKeyframe + 1;
+                addKeyPoint(
+                  'transform_' + key,
+                  animationDuration,
+                  transformStyle[key],
+                  easing,
+                  index === 0
+                );
+              });
+            });
+          } else {
+            propsInKeyframe = propsInKeyframe + 1;
+            addKeyPoint(
+              key,
+              animationDuration,
+              keyframe[key],
+              easing,
+              index === 0
```

Comment:
This function call and the one above it are really similar. You can create a partially applied function or override props in a shared object if you decide to refactor it that way.

Example:
```js
const addKeyPointWith = (key, styleProps) => addKeyPoint(
  key,
  animationDuration,
  styleProps,
  easing,
  index === 0)

// or
const baseArgs = {...} // really bad name though
addKeyPoint({...baseArgs, key, styleProp: keyframe[key]})
```